homeassistant/components/recorder/migration.py

```diff
@@ -727,6 +727,7 @@ def _add_constraint(
             connection.execute(add_constraint)
         except (InternalError, OperationalError):
             _LOGGER.exception("Could not update foreign options in %s table", table)
+            raise
```

Comment:
Any time we trap `OperationalError` from a disconnect and prevent it from raising in a session scope the server can never recover with MySQL